### PR TITLE
Fix database detection for lsvast

### PR DESCRIPTION
**Pull request by dominiklohmann**
###  :notebook_with_decorative_cover: Description

@lava feel free to push onto this branch if you immediately see the issue with the second commit. I don't think it's worth spending too much time on the matter.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

Commit-by-commit. To be reviewed by @lava.

### CODE REVIEW

File: tools/lsvast/lsvast.cpp [hunk 1]
```diff
@@ -142,10 +142,15 @@ caf::expected<Kind> classify(const std::filesystem::path& path) {
...
-
-  if (is_directory && path.stem() == "vast.db")
-    return Kind::DatabaseDir;
-
+  if (is_directory) {
+    auto is_regular_file
+      = std::filesystem::is_regular_file(path / "VERSION", err);
+    if (err)
+      return caf::make_error(vast::ec::filesystem_error,
+                             "Invalid path: ", err.message());
```
Comment (JoeLoser) [inline tools/lsvast/lsvast.cpp:150]: Should we print the path as well in the error message?

File: tools/lsvast/lsvast.cpp [hunk 2]
```diff
@@ -142,10 +142,15 @@ caf::expected<Kind> classify(const std::filesystem::path& path) {
   if (err)
     return caf::make_error(vast::ec::filesystem_error,
                            "Invalid path: ", err.message());
-
-  if (is_directory && path.stem() == "vast.db")
```
Comment (JoeLoser) [inline tools/lsvast/lsvast.cpp:146]: Did I accidentally introduce this bug when switching to `std::filesystem` here? 🤔 

Sorry if I did!

Comment (dominiklohmann) [reply tools/lsvast/lsvast.cpp:146]: You didn't, it's been broken for quite a while apparently. This is an internal debugging tool, so it's not really covered by tests or CI. We usually just work on it when we need it.

File: tools/lsvast/lsvast.cpp (hunk 1 above)
Comment (dominiklohmann) [reply tools/lsvast/lsvast.cpp:150]: Good idea.

File: tools/lsvast/lsvast.cpp (hunk 2 above)
Comment (JoeLoser) [reply tools/lsvast/lsvast.cpp:146]: Got it; that's what I figured when I was rewriting parts of it to get off of `vast::path`. Thanks for confirming.

File: tools/lsvast/lsvast.cpp [hunk 3]
```diff
@@ -141,11 +141,17 @@ caf::expected<Kind> classify(const std::filesystem::path& path) {
...
-    return Kind::DatabaseDir;
-
+                           "Invalid path: " + path.string(), err.message());
+  if (is_directory) {
+    auto version_file = path / "VERSION";
+    auto is_regular_file = std::filesystem::is_regular_file(version_file, err);
+    if (err)
+      return caf::make_error(vast::ec::filesystem_error,
+                             "Invalid path: " + version_file.string(),
```
Comment (JoeLoser) [inline tools/lsvast/lsvast.cpp:150]: Do we need the `.string()`? I thought you fixed the `fmt` specializations for `std::filesystem::path`

Comment (dominiklohmann) [reply tools/lsvast/lsvast.cpp:150]: This isn't going through {fmt} yet.

Comment (dominiklohmann) [reply tools/lsvast/lsvast.cpp:150]: For what it's worth, we do plan to do this in `vast/error.hpp` eventually:

```cpp
namespace vast {

/// Create an error with the error context formatted via `fmt::format`.
/// @param code The VAST error code.
/// @param args The arguments to forward to `fmt::format` to construct the error
/// context.
template <class... Args>
auto make_error(ec code, Args&&... args) {
  if constexpr (sizeof...(args) <= 1)
    return caf::make_error(code, std::forward<Args>(args)...);
  else
    return caf::make_error(code, fmt::format(std::forward<Args>(args)...));
}

} // namespace vast

namespace caf {

/// Poison `caf::make_error` for `vast::ec` to ensure users go through the
/// properly formatted version `vast::make_error`.
template <class... Args>
error make_error(::vast::ec, Args&&...) = delete;

} // namespace caf
```

**Comment by mavam** [comment]
@JoeLoser as a side note for your planning: we plan to migrate the functionality of `lsvast` into a dedicated command `vast dump db`. If you are up for this slightly more involved refactoring, we'd love to give you more details.

**Comment by lava** [comment]
@mavam I think you may have skipped the part where this PR was opened by @dominiklohmann , not @JoeLoser ;)

Also, our plan was to expose the functionality *in addition* to having a separate binary, since we always stressed that this is primarily a debugging tools so it's important to guarantee that `lsvast` does not modify the on-disk state (in particular does not append to `server.log`) and can still work with corrupted database dirs, which is hard to verify after we have spawned our entire regular actor system.

**Review by lava — APPROVED** [review]
As we discussed, I pushed a fix for the segfault directly onto the PR branch.

I'm approving the first part of the PR, you might want to look again over the second commit yourself.

**Comment by dominiklohmann** [comment]
Looks good to me. I've enabled auto-merge.

**Comment by mavam** [comment]
> @mavam I think you may have skipped the part where this PR was opened by @dominiklohmann , not @JoeLoser ;)

Oh boy, indeed! 🙈 

Well, @JoeLoser is contributing so much goodness that I begin to assume he's authoring most of the refactoring PRs. 😅

**Pull request merged**